singlestoredb/fusion/handlers/job.py

```diff
@@ -0,0 +1,574 @@
+    execute_every = EXECUTE EVERY <integer>
+
+    # Start time
+    start_at = START AT '<year>-<month>-<day> <hour>:<min>:<sec>'
+
+    # Resume target if suspended
+    resume_target = RESUME TARGET
+
+    # Parameters to pass to the job
+    with_parameters = WITH PARAMETERS '<parameters>'
+
+    Description
+    -----------
+    Creates a scheduled notebook job.
+
+    Arguments
+    ---------
+    * ``<notebook-path>``: The path in the Stage where the notebook file is
+      stored.
+    * ``<mode>``: The mode of the job. Either **Once** or **Recurring**.
+    * ``<runtime-name>``: The name of the runtime the job will be run with.
+    * ``<job-name>``: The name of the job.
+    * ``<job-description>``: The description of the job.
+    * ``<integer>``: The interval in minutes at which the job will be executed.
+    * ``<year>-<month>-<day> <hour>:<min>:<sec>``: The start date and time of the
+      job in UTC. The format is **yyyy-MM-dd HH:mm:ss**. The hour is in 24-hour format.
+    * ``<parameters>``: The parameters to pass to the job. A JSON string with
+      the following format:
+      ``{"parameters": [{"name": "<name>", "value": "<value>"}, ...]}``.
+
+    Remarks
+    -------
+    * The ``WITH MODE`` clause specifies the mode of the job and is either
+      **Once** or **Recurring**.
+    * The ``EXECUTE EVERY`` clause specifies the interval in minutes at which the
+      job will be executed and is required if the mode is **Recurring**.
+    * The ``CREATE SNAPSHOT`` clause creates a snapshot of the notebook executed by
+      the job.
+    * The ``WITH RUNTIME`` clause specifies the name of the runtime that
+      the job will be run with.
+    * The ``RESUME TARGET`` clause resumes the job's target if it is suspended.
+    * The ``WITH PARAMETERS`` clause specifies the parameters to pass to the job. The
+    only supported parameter value types are strings, integers, floats, and booleans.
+
+    Example
+    -------
+    The following command creates a job that will run the content of notebook
+    **example_notebook.ipynb** every 5 minutes starting at **2024-06-25 21:35:06**
+    using the runtime **notebooks-cpu-small**. The job's target will be resumed if it
+    is suspended, a snapshot of the notebook will be created and the job is named
+    **example_job** with the description **This is an example job**. The job will
+    have the following parameters: **strParam** with value **"string"**, **intParam**
+    with value **1**, **floatParam** with value **1.0**, and **boolParam** with value
+    **true**::
+
+        SCHEDULE JOB USING NOTEBOOK 'example_notebook.ipynb'
+            WITH MODE 'Recurring'
+            CREATE SNAPSHOT
+            WITH RUNTIME 'notebooks-cpu-small'
+            WITH NAME 'example_job'
+            WITH DESCRIPTION 'This is an example job'
+            EXECUTE EVERY 5
+            START AT '2024-06-25 21:35:06'
+            RESUME TARGET
+            WITH PARAMETERS '{
```

Comment:
@kesmit13 do you think this is the best way to support parameters?

Reply:
Is this the same format as used in the Management API? I think it's ok to use JSON, but I'm not fond of this format. It duplicates the `parameters` name and is quite verbose for something that would be typed by hand. Could it be changed to something like `WITH PARAMETERS '{"strParam": "string", "intParam": 1, "floatParam", 1.0, "boolParam": true}'`? That would be a lot easier to type, but may not allow other information that may be needed; for example, if it's possible to specify a data type like `{"name": "dateParam", "value": "2004-01-01", type: "date"}`, we would need a way to make that work in the simpler format.

Reply:
The management API will receive a list of tuples (name, value, type) and the python SDK will handle the type inferring so that the user does not need to input the type.
So I believe I can change to the format you suggested.